pre_commit_hooks/trailing_whitespace_fixer.py

```diff
@@ -3,17 +3,33 @@
 import argparse
 import fileinput
 import sys
+import os
 
 from pre_commit_hooks.util import cmd_output
 
 
-def _fix_file(filename):
+def _fix_file(filename, markdown=False):
     for line in fileinput.input([filename], inplace=True):
+        # preserve trailing two-space for non-blank lines in markdown files
+        if markdown and (not line.isspace()) and (line.endswith("  \n")):
+            line = line.rstrip(' \n')
+            # only preserve if there are no trailing tabs or unusual whitespace
+            if not line[-1].isspace():
+                print(line + "  ")
+                continue
+
         print(line.rstrip())
 
 
 def fix_trailing_whitespace(argv=None):
     parser = argparse.ArgumentParser()
+    parser.add_argument('--no-markdown-linebreak-ext', action='store_const',
+                        const=[], default=argparse.SUPPRESS,
+                        dest='markdown_linebreak_ext',
+                        help='Do not preserve linebreak spaces in Markdown')
+    parser.add_argument('--markdown-linebreak-ext', action='append', const='*',
+                        default=argparse.SUPPRESS, metavar='EXTS', nargs='?',
+                        help='Markdown extensions (or *) for linebreak spaces')
```

Comment:
I think we can actually simplify this to one argument:

```
parser.add_argument(
    '--markdown-trailing-whitespace-extension',
    action='append',
    help='...',
)
```

The default (empty list) would mean nothing is treated as markdown, you would add more extensions by specifying the argument more than once, and the string `'*'` would be the special case where all are treated as markdown (though why support this? I'm not sure anyone will use the `'*'` option?)